app/src/main/java/emu/skyline/views/CustomLinearLayout.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: MPL-2.0
+ * Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
+ */
+
+package emu.skyline.views
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.LinearLayout
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import com.google.android.material.snackbar.Snackbar.SnackbarLayout
+
+class CustomLinearLayout : LinearLayout, CoordinatorLayout.AttachedBehavior {
+
+    constructor(context : Context) : this(context, null)
+    constructor(context : Context, attrs : AttributeSet?) : this(context, attrs, 0)
+    constructor(context : Context, attrs : AttributeSet?, defStyleAttr : Int) : super(context, attrs, defStyleAttr)
+
+    override fun getBehavior() : CoordinatorLayout.Behavior<CustomLinearLayout> = MoveUpwardBehavior()
+
+    class MoveUpwardBehavior : CoordinatorLayout.Behavior<CustomLinearLayout>() {
+        override fun layoutDependsOn(parent : CoordinatorLayout, child : CustomLinearLayout, dependency : View) : Boolean {
+            return dependency is SnackbarLayout
+        }
+
+        override fun onDependentViewChanged(parent : CoordinatorLayout, child : CustomLinearLayout, dependency : View) : Boolean {
+            val translationY = .0f.coerceAtMost(dependency.translationY - dependency.height)
+            child.translationY = translationY
```

Comment:
You can avoid having the extra variable by directly assigning it to `child.translationY`. I'm assuming you did this for readability but due to the assigned statement already being small enough there isn't a huge point. In addition, `.0f` looks a bit off when combined with a statement, I think doing `(0.0f).coerceAtMost` is far more clear.